util/internal/integrationtest/mapper/orm/src/main/java/org/hibernate/search/util/impl/integrationtest/mapper/orm/OrmSetupHelper.java

```diff
@@ -60,53 +65,57 @@ public final class OrmSetupHelper
 	}
 
 	public static OrmSetupHelper withBackendMock(BackendMock backendMock) {
-		return withBackendMock( Type.METHOD, backendMock );
+		return withBackendMock( Type.METHOD, backendMock, JUPITER_ASSUMPTION_CHECK );
 	}
 
 	public static OrmSetupHelper withBackendMocks(BackendMock defaultBackendMock,
 			Map<String, BackendMock> namedBackendMocks) {
-		return withBackendMocks( Type.METHOD, defaultBackendMock, namedBackendMocks );
+		return withBackendMocks( Type.METHOD, defaultBackendMock, namedBackendMocks, JUPITER_ASSUMPTION_CHECK );
 	}
 
 	public static OrmSetupHelper withSingleBackend(BackendConfiguration backendConfiguration) {
-		return withSingleBackend( Type.METHOD, backendConfiguration );
+		return withSingleBackend( Type.METHOD, backendConfiguration, JUPITER_ASSUMPTION_CHECK );
 	}
 
 	public static OrmSetupHelper withMultipleBackends(BackendConfiguration defaultBackendConfiguration,
 			Map<String, BackendConfiguration> namedBackendConfigurations) {
-		return withMultipleBackends( Type.METHOD, defaultBackendConfiguration, namedBackendConfigurations );
+		return withMultipleBackends( Type.METHOD, defaultBackendConfiguration, namedBackendConfigurations,
+				JUPITER_ASSUMPTION_CHECK );
 	}
 
 	public static OrmSetupHelper withBackendMockGlobal(BackendMock backendMock) {
-		return withBackendMock( Type.CLASS, backendMock );
+		return withBackendMock( Type.CLASS, backendMock, VINTAGE_ASSUMPTION_CHECK );
```

Comment:
If this is about JUnit 4 vs. JUnit 5, you might want to use a different suffix for this method than "Global"... ?

Reply:
Once we replace the `ReusableOrmSetupHolder` all of this will go away, for now I've renamed it to "legacy" 😃